library/tests/src/lib.rs

```diff
@@ -34,5 +38,37 @@ pub fn test_expression(expr: &str, expected: &Value) {
         .eval_entry(&mut out)
         .expect("test should run successfully");
 
-    assert_eq!(expected, &result);
+    match (&expected, result) {
+        (&Value::Tuple(tup1), Value::Tuple(tup2)) if tup1.len() == tup2.len() => {
+            // If both values are tuples of the same length, we crack them open and compare elements
+            for (value1, value2) in tup1.iter().zip(tup2.iter()) {
+                if let (Value::Double(double1), Value::Double(double2)) = (value1, value2) {
+                    // If both elements are doubles, we use approximate comparison
+                    assert_doubles_almost_equal(*double1, *double2);
```

Comment:
One minor note that I hadn't thought of before: this will only work one "layer" deep, so if the data structure under test is a tuple of tuple of double, as in `((Double, Double), Double)`, the inner tuple will not get caught here and still go through direct comparison. That isn't a problem right now with how this is used, but something we might want to keep in mind for the future.

Reply:
True. Also we don't handle array elements, etc. I guess we will exert the effort when we need it :)